crates/driver/src/infra/api/quote/dto/quote.rs

```diff
@@ -0,0 +1,119 @@
+            buy_token: quote.buy.token.into(),
+            sell_amount: quote.sell.amount,
+            buy_amount: quote.buy.amount,
+            interactions: quote
+                .interactions
+                .iter()
+                .map(|interaction| match interaction {
+                    competition::solution::Interaction::Custom(interaction) => {
+                        Interaction::Custom(CustomInteraction {
+                            internalize: interaction.internalize,
+                            target: interaction.target.into(),
+                            value: interaction.value.into(),
+                            call_data: interaction.call_data.clone(),
+                            inputs: interaction
+                                .inputs
+                                .iter()
+                                .map(|asset| Asset {
+                                    token: asset.token.into(),
+                                    amount: asset.amount,
+                                })
+                                .collect(),
+                            outputs: interaction
+                                .outputs
+                                .iter()
+                                .map(|asset| Asset {
+                                    token: asset.token.into(),
+                                    amount: asset.amount,
+                                })
+                                .collect(),
+                        })
+                    }
+                    competition::solution::Interaction::Liquidity(interaction) => {
+                        Interaction::Liquidity(LiquidityInteraction {
+                            internalize: interaction.internalize,
+                            id: interaction.liquidity.id.into(),
+                            input_token: interaction.input.token.into(),
+                            output_token: interaction.output.token.into(),
+                            input_amount: interaction.input.amount,
+                            output_amount: interaction.output.amount,
+                        })
+                    }
+                })
+                .collect(),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+#[serde_as]
+#[serde(rename_all = "camelCase")]
+pub struct Quote {
+    // TODO I think this might be a mistake. Instead of these four fields, how about this just
+    // returns a single amount field which is a U256? That should be enough, right? It would
+    // also simplify some of the other code which is nice.
+    sell_token: eth::H160,
+    buy_token: eth::H160,
+    #[serde_as(as = "serialize::U256")]
+    sell_amount: eth::U256,
+    #[serde_as(as = "serialize::U256")]
+    buy_amount: eth::U256,
+
+    interactions: Vec<Interaction>,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "kind", rename_all = "lowercase")]
+enum Interaction {
+    Liquidity(LiquidityInteraction),
+    Custom(CustomInteraction),
+}
```

Comment:
This DTO doesn't make sense to me, as the `driver` will always provide "raw" interactions to the caller (this looks like the interaction DTO between `driver`<>`solver`). In fact, the liquidity is fetched by the driver, so `LiquidityInteraction` won't really be usable by the `autopilot`/`orderbook`.

Specifically, I think this should be:
```rust
struct Interaction {
    target: eth::H160,
    value: eth::EtherValue,
    calldata: Vec<u8>,
}
```